sbnci/Modules/TrackValidation/TrackValidation_module.cc

```diff
@@ -0,0 +1,3127 @@
+                                                        std::map<int,float>& MCTrackEnergyMap){
+  //###### Declaration and Initiate variables #######
+  for(auto const& trkmother : TracksMothers){
+    
+    //Loop over the daughter
+    float deposited_energy = 0.f;
+    float sim_energy       = (trueParticlesMap[trkmother.first]->E())*1000;
+    for(auto const& daughter : trkmother.second){
+      deposited_energy += MCTrackEnergyMap[daughter];
+    }
+    //If the energy of track is over 90% seen on the wires in truth, It is contained.
+    if((deposited_energy/sim_energy) < 0.9){
+      TracksMothers.erase(trkmother.first);
+    }
+  }
+  return;
+}//Close the function that remove non contained particles
+
+//Function cut on the Track mothers with energy
+void ana::TrackValidation::CutTrkMothersByE(std::map<int,std::vector<int> >&  TracksMothers,
+                                            std::map<int,const simb::MCParticle*>& trueParticlesMap,
+                                            float& EnergyCut){
+  //Cut the true tracks and make sure they are a tracks.
+  for(auto const& mapiter : TracksMothers){
+    const simb::MCParticle* TrkMotherParticle = trueParticlesMap[mapiter.first];
+    if(TrkMotherParticle->E() < EnergyCut){
+      TracksMothers.erase(mapiter.first);
+    }
+  }//close the loop over mothers
+  return;
+}//close the function
+
+//Function for cut on track via length 
+void ana::TrackValidation::CalculateTrackLengthCut(std::map<int,std::vector<int> >&  TracksMothers,
+                                                   std::map<int,const simb::MCParticle*>& trueParticlesMap,
+                                                   float& LengthCut){
+  //###### Declaration and Initiate variables #######
+  for(auto const& mapiter : TracksMothers){
+    float TrackLength = 0.f;
+    bool startpc = false;
+    bool endtpc  = false;
+    const simb::MCParticle* TrkMotherParticle = trueParticlesMap[mapiter.first];
+    //Get the number of Traj points to loop over
+    unsigned int TrajPoints = TrkMotherParticle->NumberTrajectoryPoints();
+    // Select first traj point where the track loses energy, last be default
+    TLorentzVector PositionTrajStart =  TrkMotherParticle->Position(0);
+    TLorentzVector PositionTrajEnd   =  TrkMotherParticle->Position(TrajPoints-1);
+    //Check if the start and end position of track is inside the tpc
+    startpc = IsInsideTPC((PositionTrajStart.Vect()), 0);
+    endtpc  = IsInsideTPC((PositionTrajEnd.Vect()), 0);
+    //Get the track length.
+    TrackLength = (PositionTrajStart-PositionTrajEnd).Vect().Mag();
+    if((!startpc) || (!endtpc) || (TrackLength < LengthCut)){
+      TracksMothers.erase(mapiter.first);
+    }
+  }//close the loop over tracks mothers
+  return;
+}//close the function 
+
+//Function return a bool; true if the track inside the tpc and false if not.
+bool ana::TrackValidation::IsInsideTPC(TVector3 position, double distance_buffer){
```

Comment:
Does this differ from the one in RecoUtils? https://github.com/SBNSoftware/sbnci/blob/0a0b710a85d9970e37d8c15cf8539dd2cbee88b5/sbnci/Modules/MCRecoUtils/RecoUtils.h#L51

Reply:
resolved